### ISSUE TITLE
Propfind returns 400 Bad request

### ISSUE THREAD

**Issue by NLLAPPS** [thegrizzlylabs/sardine-android#1]
Hi and thanks for the port.
I have been trying to implement it but there seems to be an issue with getResurces()

I have tested with 3 different servers, all return 400 when you call it
`  List<DavResource> list = getResources(url);`

**Comment by guillaume-tgl** [comment]
Hi, I've just made a commit with lots of fixes. Could you try again?

**Comment by NLLAPPS** [comment]
Just off the computer, will do as soon as possible and report back

**Comment by NLLAPPS** [comment]
I think something wrong with result parsing. I use  propfind to check if file/path exist because on at least Synology Diskstations making a request to uri to find out if exists doe not work. You get 403 (i guess it is related to preemptive authentication as it happens if i don't enable it when using Apache jackrabbit with Apache http client)

Anyway, with latest updates request is successful but parsing fails, here is the Exception

`11-24 19:00:56.845 9082-10722/? W/System.err: java.io.IOException: Not a valid DAV response
11-24 19:00:56.845 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:174)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.MultiStatusResponseHandler.handleResponse(MultiStatusResponseHandler.java:3038)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.ResourcesResponseHandler.handleResponse(ResourcesResponseHandler.java:25)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.ResourcesResponseHandler.handleResponse(ResourcesResponseHandler.java:19)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.execute(OkHttpSardine.java:613)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.propfind(OkHttpSardine.java:158)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.exists(OkHttpSardine.java:6127)
11-24 19:00:56.846 9082-10722/? W/System.err:     at com.nll.cloud.webdav.SardineWebDAV.exists(SardineWebDAV.java:149)
`

**Comment by NLLAPPS** [comment]
Well, apparently checking existence of an a WebDAV path by simply a request is fixed with your latest update. So, there is no need to use propfind in my case. But, issue above is still present when propfind (getResources(uri)) used

**Comment by NLLAPPS** [comment]
By the way I am simply calling getResources(url)

**Comment by guillaume-tgl** [comment]
Could you paste entire stacktrace as the one you gave is missing the most important :) ?

**Comment by NLLAPPS** [comment]
Full stack trace

`11-27 09:01:17.523 31897-32117/com.test I/System.out: Authenticating for response: Response{protocol=http/1.1, code=401, message=Authorization Required, url=http://192.168.0.78:5005/WEBDAV_TEST/TestFolder}
11-27 09:01:17.524 31897-32117/com.test I/System.out: Challenges: [Basic realm="SYNO_WebDAV Storage" charset="ISO-8859-1"]
11-27 09:01:17.636 31897-32117/com.test W/System.err: java.io.IOException: Not a valid DAV response
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:174)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.MultiStatusResponseHandler.getMultistatus(MultiStatusResponseHandler.java:38)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.MultiStatusResponseHandler.handleResponse(MultiStatusResponseHandler.java:27)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.ResourcesResponseHandler.handleResponse(ResourcesResponseHandler.java:25)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.handler.ResourcesResponseHandler.handleResponse(ResourcesResponseHandler.java:19)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.execute(OkHttpSardine.java:612)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.propfind(OkHttpSardine.java:157)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.list(OkHttpSardine.java:126)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.list(OkHttpSardine.java:102)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.list(OkHttpSardine.java:97)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.getResources(OkHttpSardine.java:92)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.exists(OkHttpSardine.java:360)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.nll.cloud.webdav.SardineWebDAV.exists(SardineWebDAV.java:149)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.nll.cloud.webdav.WebDavClient.checkAndCreateFolders(WebDavClient.java:156)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.nll.cloud.webdav.WebDavClient.init(WebDavClient.java:64)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.nll.cloud.webdav.WebDAVConnectionTest.doInBackground(WebDAVConnectionTest.java:25)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at com.nll.cloud.webdav.WebDAVConnectionTest.doInBackground(WebDAVConnectionTest.java:10)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at android.os.AsyncTask$2.call(AsyncTask.java:305)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
11-27 09:01:17.637 31897-32117/com.test W/System.err:     at java.lang.Thread.run(Thread.java:762)
11-27 09:01:17.638 31897-32117/com.test W/System.err: Caused by: org.simpleframework.xml.core.PersistenceException: Constructor not matched for class com.thegrizzlylabs.sardineandroid.model.Collection
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.ClassInstantiator.getInstance(ClassInstantiator.java:112)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.readInject(Composite.java:1458)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.read(Composite.java:1435)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.read(Composite.java:1433)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 09:01:17.638 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:190)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:167)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:124)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:190)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:167)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:124)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:625)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:606)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:584)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:543)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:444)
11-27 09:01:17.639 31897-32117/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:169)
11-27 09:01:17.639 31897-32117/com.test W/System.err: 	... 22 more
`

**Comment by guillaume-tgl** [comment]
I can't reproduce the issue. Could you either:
- give me the response from the server (you can set a break point in `OkHttpSardine.execute` and watch `response.body().source().readUtf8()`
- or open an access on your WebDAV server so that I can debug directly on it?

**Comment by NLLAPPS** [comment]
getResources Request to http://192.168.0.78:5005/WEBDAV_TEST/TestFolder

returns 
`<?xml version="1.0" encoding="utf-8"?>
<D:multistatus xmlns:D="DAV:" xmlns:ns0="DAV:">
<D:response xmlns:lp2="http://apache.org/dav/props/" xmlns:lp1="DAV:">
<D:href>/WEBDAV_TEST/TestFolder/</D:href>
<D:propstat>
<D:prop>
<lp1:resourcetype><D:collection/></lp1:resourcetype>
<lp1:creationdate>2017-11-27T13:59:55Z</lp1:creationdate>
<lp1:getlastmodified>Mon, 27 Nov 2017 13:59:55 GMT</lp1:getlastmodified>
<lp1:getetag>"8d1021-1000-55ef7513505d5"</lp1:getetag>
<D:supportedlock>
<D:lockentry>
<D:lockscope><D:exclusive/></D:lockscope>
<D:locktype><D:write/></D:locktype>
</D:lockentry>
<D:lockentry>
<D:lockscope><D:shared/></D:lockscope>
<D:locktype><D:write/></D:locktype>
</D:lockentry>
</D:supportedlock>
<D:lockdiscovery/>
<D:getcontenttype>httpd/unix-directory</D:getcontenttype>
</D:prop>
<D:status>HTTP/1.1 200 OK</D:status>
</D:propstat>
</D:response>
</D:multistatus>`

error prints

`java.io.IOException: Not a valid DAV response
at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:174)`

**Comment by guillaume-tgl** [comment]
I still can't reproduce this issue. I tried to run the following deserialization and it worked without error: 
`Multistatus multistatus = SardineUtil.unmarshal(Multistatus.class, new ByteArrayInputStream(<your xml response>.getBytes()));`

Are you sure you have the latest code?
If you can reproduce this in a small project, I'd gladly have a look at it.

**Comment by NLLAPPS** [comment]
I have added e.printStackTrace to SardineUtil.unmarshal(Multistatus.class, stream) and this is what it printed

`11-27 15:35:36.064 19759-20124/com.test W/System.err: org.simpleframework.xml.core.PersistenceException: Constructor not matched for class com.thegrizzlylabs.sardineandroid.model.Collection
11-27 15:35:36.064 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.ClassInstantiator.getInstance(ClassInstantiator.java:112)
11-27 15:35:36.064 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.readInject(Composite.java:1458)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.read(Composite.java:1435)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Injector.read(Composite.java:1433)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 15:35:36.065 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:190)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:167)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:124)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:190)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:167)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:124)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
11-27 15:35:36.066 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:201)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Composite.read(Composite.java:148)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:625)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:606)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:584)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:543)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at org.simpleframework.xml.core.Persister.read(Persister.java:444)
11-27 15:35:36.067 19759-20124/com.test W/System.err:     at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:169)`

**Comment by NLLAPPS** [comment]
This is also with your latest commit (from around 10 minutes ago)

**Comment by guillaume-tgl** [comment]
Funny, I can still not reproduce.

Here is my exact code: 
```
@Test
public void unmarshall() throws IOException {
    String xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?> <D:multistatus xmlns:D=\"DAV:\" xmlns:ns0=\"DAV:\"> <D:response xmlns:lp2=\"http://apache.org/dav/props/\" xmlns:lp1=\"DAV:\"> <D:href>/WEBDAV_TEST/TestFolder/</D:href> <D:propstat> <D:prop> <lp1:resourcetype><D:collection/></lp1:resourcetype> <lp1:creationdate>2017-11-27T13:59:55Z</lp1:creationdate> <lp1:getlastmodified>Mon, 27 Nov 2017 13:59:55 GMT</lp1:getlastmodified> <lp1:getetag>\"8d1021-1000-55ef7513505d5\"</lp1:getetag> <D:supportedlock> <D:lockentry> <D:lockscope><D:exclusive/></D:lockscope> <D:locktype><D:write/></D:locktype> </D:lockentry> <D:lockentry> <D:lockscope><D:shared/></D:lockscope> <D:locktype><D:write/></D:locktype> </D:lockentry> </D:supportedlock> <D:lockdiscovery/> <D:getcontenttype>httpd/unix-directory</D:getcontenttype> </D:prop> <D:status>HTTP/1.1 200 OK</D:status> </D:propstat> </D:response> </D:multistatus>";
    SardineUtil.unmarshal(Multistatus.class, new ByteArrayInputStream(xml.getBytes()));
}
```

Can you also try to add a no-arg constructor to the `Collection` class to see if it helps?

**Comment by NLLAPPS** [comment]
Well! 
`String xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?> <D:multistatus xmlns:D=\"DAV:\" xmlns:ns0=\"DAV:\"> <D:response xmlns:lp2=\"http://apache.org/dav/props/\" xmlns:lp1=\"DAV:\"> <D:href>/WEBDAV_TEST/TestFolder/</D:href> <D:propstat> <D:prop> <lp1:resourcetype><D:collection/></lp1:resourcetype> <lp1:creationdate>2017-11-27T13:59:55Z</lp1:creationdate> <lp1:getlastmodified>Mon, 27 Nov 2017 13:59:55 GMT</lp1:getlastmodified> <lp1:getetag>\"8d1021-1000-55ef7513505d5\"</lp1:getetag> <D:supportedlock> <D:lockentry> <D:lockscope><D:exclusive/></D:lockscope> <D:locktype><D:write/></D:locktype> </D:lockentry> <D:lockentry> <D:lockscope><D:shared/></D:lockscope> <D:locktype><D:write/></D:locktype> </D:lockentry> </D:supportedlock> <D:lockdiscovery/> <D:getcontenttype>httpd/unix-directory</D:getcontenttype> </D:prop> <D:status>HTTP/1.1 200 OK</D:status> </D:propstat> </D:response> </D:multistatus>";
    SardineUtil.unmarshal(Multistatus.class, new ByteArrayInputStream(xml.getBytes()));`

fails on me! Can it be proguard?

I am using org.simpleframework:simple-xml:2.7.1  with

`-keep class org.simpleframework.** {*;}
-keep class org.xmlpull.** { *; }
-dontwarn org.xmlpull.v1.**
-dontwarn javax.xml.namespace.QName`

**Comment by NLLAPPS** [comment]
Target is 27 and min sdk is 14 so, they seems to be fine and your simple-xml is 2.7.1, what can it be?

**Comment by NLLAPPS** [comment]
However below works just fine

`<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
<d:response><d:href>/remote.php/webdav/test/</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 27 Nov 2017 20:04:35 GMT</d:getlastmodified><d:resourcetype><d:collection/></d:resourcetype><d:quota-used-bytes>0</d:quota-used-bytes><d:quota-available-bytes>-3</d:quota-available-bytes><d:getetag>&quot;5a1c6fd363154&quot;</d:getetag></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>
`

**Comment by NLLAPPS** [comment]
Solved!

progaurd was the culprit!

adding `-keep class com.thegrizzlylabs.sardineandroid.** {*;}` fixes it

What I don't understand is why below ha no issues
`<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
<d:response><d:href>/remote.php/webdav/test/</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 27 Nov 2017 20:04:35 GMT</d:getlastmodified><d:resourcetype><d:collection/></d:resourcetype><d:quota-used-bytes>0</d:quota-used-bytes><d:quota-available-bytes>-3</d:quota-available-bytes><d:getetag>&quot;5a1c6fd363154&quot;</d:getetag></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>`

**Comment by guillaume-tgl** [comment]
I've just added Proguard rules to the library. Let me know if we can close this.
